x/mongo/driver/auth/creds/aws_credential_providers/ecs_provider.go

```diff
@@ -0,0 +1,104 @@
+// Copyright (C) MongoDB, Inc. 2023-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package awscredproviders
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"go.mongodb.org/mongo-driver/x/mongo/driver/auth/internal/aws/credentials"
+)
+
+const (
+	// EcsProviderName provides a name of ECS provider
+	EcsProviderName = "EcsProvider"
```

Comment:
This constant has no usages outside of the `awscredproviders` package, so does not need to be exported.